src/Passes/Source/ValidationPass/ValidationPass.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Logging/ILogger.hpp"
+#include "Profile/Profile.hpp"
+#include "Types/Types.hpp"
+#include "ValidationPass/ValidationConfiguration.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <functional>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class ValidationPass : public llvm::PassInfoMixin<ValidationPass>
+    {
+      public:
+        using Instruction       = llvm::Instruction;
+        using Value             = llvm::Value;
+        using ConstantArguments = std::unordered_map<std::string, llvm::ConstantInt*>;
+        using ILoggerPtr        = std::shared_ptr<ILogger>;
```

Comment:
I have an impression that some of these types are not used anywhere, in particular `ILoggerPtr`, `ConstantArguments`.

Reply:
Removed.